src/JMS/Serializer/GenericSerializationVisitor.php

```diff
@@ -162,12 +162,23 @@ public function visitProperty(PropertyMetadata $metadata, $data, Context $contex
      */
     public function addData($key, $value)
     {
-        if (isset($this->data[$key])) {
+        if ( $this->hasData($key) ) {
```

Comment:
and I think the isset check could still be done directly here rather than adding an extra method call